lib/ansible/cli/galaxy.py

```diff
@@ -969,39 +991,115 @@ def _display_role(gr):
+                        if path in C.COLLECTIONS_PATHS:
+                            # don't warn for missing default dirs
+                            continue
+                        warnings.append("- the configured path {0} does not exist.".format(collection_path))
+                        continue
+                    elif not os.path.isdir(collection_path):
+                        warnings.append("- the configured path {0}, exists, but it is not a directory.".format(collection_path))
+                        continue
+
+                    path_found = True
+                    display.display("# {0}".format(path))
+
+                    # Make sure we look inside the 'ansible_collections' dir
+                    if os.path.split(collection_path)[1] != 'ansible_collections':
+                        collection_path = os.path.join(collection_path, 'ansible_collections')
+
+                    try:
+                        namespace, collection = collection_name.split('.')
+                    except ValueError:
+                        raise AnsibleOptionsError("Incorrect value for collection name. Must be 'namespace.collection' got '%s'" % collection_name)
+
+                    b_collection_path = to_bytes(os.path.join(collection_path, namespace, collection), errors='surrogate_or_strict')
+
+                    if not os.path.exists(b_collection_path):
+                        warnings.append("- unable to find {0} in collection paths".format(collection_name))
+                        continue
+
+                    collection = CollectionRequirement.from_path(b_collection_path, False)
+                    _display_collection(collection)
+
+            else:
+                # list all collections
+                for path in collections_search_paths:
+                    collection_path = GalaxyCLI._resolve_path(path)
+                    if not os.path.exists(path):
+                        if path in C.COLLECTIONS_PATHS:
+                            # don't warn for missing default dirs
+                            continue
+                        warnings.append("- the configured path {0} does not exist.".format(collection_path))
+                        continue
+                    elif not os.path.isdir(collection_path):
+                        warnings.append("- the configured path {0}, exists, but it is not a directory.".format(collection_path))
+                        continue
+
+                    display.display("# {0}".format(path))
+
+                    # Make sure we look inside the 'ansible_collections' dir
+                    if os.path.split(path)[1] != 'ansible_collections':
+                        path = os.path.join(path, 'ansible_collections')
+
+                    collections = _find_existing_collections(path)
+                    # Sort collections by the namespace and name
+                    collections.sort(key=lambda x: '%s.%s' % (x.namespace, x.name))
```

Comment:
The object's `repr` has the string I want, `namespace.collection`. Is there a way to use that and sort based on it rather than having to use a `lambda` here?